janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -5685,6 +5694,36 @@ public void testNotHas() {
         assertTrue(tx.traversal().V().not(__.has("p3")).hasNext());
     }
 
+    @Test
+    public void testVertexCentricQueryProfiling() {
+        final PropertyKey time = mgmt.makePropertyKey("time").dataType(Integer.class).cardinality(Cardinality.SINGLE).make();
+        final EdgeLabel friend = mgmt.makeEdgeLabel("friend").multiplicity(Multiplicity.SIMPLE).make();
+        mgmt.buildEdgeIndex(friend, "byTime", OUT, desc, time);
+        finishSchema();
+
+        JanusGraphVertex v = tx.addVertex();
+        JanusGraphVertex o = tx.addVertex();
+        v.addEdge("friend", o, "time", 100);
+        v.addEdge("friend-no-index", o, "time", 100);
+        tx.commit();
+
+        // satisfied by a single vertex-centric query which is satisfied by one edge query
+        newTx();
+        Metrics mSingleLabel = tx.traversal().V(v).outE("friend").has("time", P.lt(10)).profile().next().getMetrics(1);
+        assertEquals("JanusGraphVertexStep([time.lt(10)])", mSingleLabel.getName());
+        assertTrue(mSingleLabel.getDuration(TimeUnit.MICROSECONDS) > 0);
+        Map<String, String> annotations = new HashMap() {{
```

Comment:
mit: Wouldn't be better to ensure the order of annotations? I'm not sure if this is possible.

Reply:
It should be technically viable, although non-trivial to compare the order of LinkedHashMaps.
My concern is, https://github.com/apache/tinkerpop/blob/master/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/ImmutableMetrics.java#L45 provides annotations as a Map, so no order is guaranteed. Although its implementation relies on LinkedHashMap which preserves insertion order, we may not want to rely on it in our test.

Reply:
I don't we should rely on the internal implementation of tinkerpop, if they just return a hashmap. It don't it is critical, all of them are basically just arguments.